source/tools/glkernel-cli/helper.cpp

```diff
@@ -0,0 +1,167 @@
+#include "helper.h"
+
+#include <algorithm>
+#include <vector>
+
+#include <glkernel/Kernel.h>
+
+void throwIfNot(bool condition, const std::string& msg)
+{
+    if (!condition)
+    {
+        throw std::logic_error(msg);
+    }
+}
+
+std::pair<float, float> findMinMaxElements(const glkernel::tkernel<float> & kernel)
+{
+  const auto minmax = std::minmax_element(kernel.cbegin(), kernel.cend());
+  return std::make_pair(*minmax.first, *minmax.second);
+}
+
+std::pair<float, float> findMinMaxElements(const glkernel::tkernel<glm::vec2> & kernel)
+{
+  float min = std::numeric_limits<float>::max();
+  float max = std::numeric_limits<float>::min();
+  const auto end = kernel.cend();
+  for (auto it = kernel.cbegin(); it != end; ++it)
```

Comment:
I suggest adding a const-overloaded `begin` and `end` to `tkernel` and use a range-based for loop instead:
```cpp
for (const auto & p : kernel)
{
    const auto x = p.x;
    const auto y = p.y;

    //...
}
```

Reply:
Done